app/core/DB/PDF2db_v2.py

```
@@ -31,22 +38,403 @@
+
+    # 긴 텍스트부터 매칭 (짧은 것이 긴 것의 부분문자열일 수 있으므로)
+    sorted_headings = sorted(heading_texts, key=len, reverse=True)
+
+    for heading in sorted_headings:
+        escaped = re.escape(heading)
+        # 공백 1개 이상이면 매칭 (pymupdf4llm이 공백을 1개로 합치는 경우 대응)
+        pattern = re.compile(
+            rf"^({escaped})\s+(\S)",
+            re.MULTILINE,
+        )
+        md_text = pattern.sub(rf"### \1\n\n\2", md_text)
```

Ruff (0.15.9)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)